.github/workflows/release-workflow.yml

```diff
@@ -0,0 +1,111 @@
+name: Release workflow
+# This workflow is triggered on creating tags to master or a opendistro release branch
+on:
+  push:
+    branches:
```

Comment:
Question about this setup, is this:
1. Triggering this workflow for all push events to the below branches or matching tags
or
2. Triggering this workflow for all push events that match the tags in only these branches

We definitely don't want #1, and concerned that might be the case here.